### PR TITLE
Prevent tensorflow and torch from being imported with PennyLane due to QNN module

### PR DESCRIPTION
**Pull request by trbromley**
**Context:**

TF and Torch are imported automatically with PL which can take a few seconds.

**Description of the Change:**

Removes import of `qnn` module from main `init`.

**Benefits:**

- Quicker import of PL
- No obscure error messages due to TF:
```
2021-03-24 17:42:34.639127: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory
2021-03-24 17:42:34.639155: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
```

**Timing**
Running `time python -c "import pennylane"` on:
- `master`:
```
real    0m3.776s
user    0m3.956s
sys     0m1.854s
```
- this branch:
```
real    0m2.099s
user    0m2.426s
sys     0m1.696s
```
Still not as fast as I expected :thinking:

**Possible Drawbacks:**

This is a breaking change. Any users with code:

```python
import pennylane as qml

qml.qnn.KerasLayer(...)
```
will have to change to
```python
import pennylane as qml
import pennylane.qnn

qml.qnn.KerasLayer(...)
```
This is a commonly-used feature so we may see users posting questions on our forum.

Demos will also need to be updated.

**Related GitHub Issues:**

Fixes https://github.com/PennyLaneAI/pennylane/issues/1101.

### CODE REVIEW

**Comment by josh146** [comment]
It definitely isn't as fast as speedup as I was expecting 🤔 We should try and profile import time

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -510,6 +512,16 @@
...
+* The `qml` module is no longer automatically imported with PennyLane and should instead be imported
+  separately using `import pennylane.qnn`. For example, `KerasLayer` can be accessed using:
+  
+  ```python
+  import pennylane as qml
+  import pennylane.qnn
+  
+  qlayer = qml.qnn.KerasLayer(...)
+  ```
+
```
Comment (trbromley) [inline .github/CHANGELOG.md:527]: To confirm - we don't want to update mentions of `qml.qnn` in the changelog before this release?

Comment (josh146) [reply .github/CHANGELOG.md:527]: nope, changelog versions prior to v0.15-dev are immutable

**Comment by josh146** [comment]
@trbromley this is only a sample size of 1, but this is the difference on my laptop.

`master`:

```python
>>> In [1]: %timeit -n 1 -r 1 import pennylane
2021-03-25 20:41:26.482992: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory
2021-03-25 20:41:26.483610: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
11.7 s ± 0 ns per loop (mean ± std. dev. of 1 run, 1 loop each)
```

`fix_qnn_imports`:

```python
>>> %timeit -n 1 -r 1 import pennylane
4.81 s ± 0 ns per loop (mean ± std. dev. of 1 run, 1 loop each)
```

**Comment by trbromley** [comment]
Wow, it took 11 seconds on your laptop! TensorFlow needs to improve on that front. But still, I wonder why we get 5 seconds without TF and torch. What else could be slowing us down? :thinking:

**Comment by josh146** [comment]
> What else could be slowing us down? 🤔

Yeah, a good thing for us to try and understand

**Comment by trbromley** [comment]
I'm not sure why it's failing with `codecov`. It is [saying](https://app.codecov.io/gh/PennyLaneAI/pennylane/compare/1162/tree/pennylane/qnn/keras.py) that the `__str__` method is not covered, but it is in the [tests](https://github.com/PennyLaneAI/pennylane/blob/0ee9411075fab09965e9e995439cb004dde08b48/tests/qnn/test_keras.py#L427) :thinking:.

**Comment by Thenerdstation** [comment]
Is JAX still being imported? That might be the large overhead we still see.

**Review by josh146 — APPROVED** [review]

**Comment by trbromley** [comment]
> Is JAX still being imported? That might be the large overhead we still see.

I just tried timing without JAX installed, but the numbers are similar (staying on this branch):
```python
real    0m2.172s
user    0m2.403s
sys     0m1.777s
```
So it looks like either JAX isn't being imported or it has little effect.

**Comment by trbromley** [comment]
Update: yes I timed importing JAX and it's very quick!
```
real    0m0.040s
user    0m0.020s
sys     0m0.021s
```

Also, these are the times importing PL with a fresh environment but without qchem installed (on the current branch):
```
real    0m0.682s
user    0m1.018s
sys     0m1.690s
```
Same environment with qchem installed:
```
real    0m2.286s
user    0m2.485s
sys     0m1.831s
```
Is there a heavy dependency in qchem?

**Comment by Thenerdstation** [comment]
> Is there a heavy dependency in qchem?

PySCF?

**Comment by trbromley** [comment]
Update: let's consider adding a deprecation warning first (https://github.com/PennyLaneAI/pennylane/pull/1170) and then merging this in after the next release.

**Comment by josh146** [comment]
@trbromley I believe this can now be closed

**Comment by trbromley** [comment]
Yes thanks @josh146, closed because we'll do https://github.com/PennyLaneAI/pennylane/pull/1228 instead.

**Pull request closed without merging**